### DISCUSSION TITLE
Installing Tuta Mail & Calendar

### DISCUSSION THREAD

**Discussion by Alexanderkocha** [ImranR98/Obtainium#2040] [Q&A] [answered]
I've been trying to get the right files to download, but I've no idea what fields to fill in and how.

The [link](https://github.com/tutao/tutanota/releases) to the Tuta repo

Would anyone be able to guide me on downloading those apps?

**Comment by DarkCrypt** [comment, marked as answer]
This has been discussed before. There's a few different ways to retrieve these 2 apps but I prefer using the APK filter.

A config for Tuta Mail which uses the APK filter and the Version Trim can be found on Obtainium's config website, https://apps.obtainium.imranr.dev/. I'll provide a config for both apps including a breakdown so you can see how it works.

Tuta Mail config (same as config website):
Import App -> [Tuta](https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22de.tutao.tutanota%22%2C%22url%22%3A%22https%3A%2F%2Fgithub.com%2Ftutao%2Ftutanota%22%2C%22author%22%3A%22tutao%22%2C%22name%22%3A%22Tuta%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22includePrereleases%5C%22%3Afalse%2C%5C%22fallbackToOlderReleases%5C%22%3Atrue%2C%5C%22filterReleaseTitlesByRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22filterReleaseNotesByRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22verifyLatestTag%5C%22%3Afalse%2C%5C%22dontSortReleasesList%5C%22%3Afalse%2C%5C%22useLatestAssetDateAsReleaseDate%5C%22%3Afalse%2C%5C%22releaseTitleAsVersion%5C%22%3Afalse%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22%5B0-9%5D%2B%5C%5C%5C%5C.%5B0-9%5D%2B%5C%5C%5C%5C.%5B0-9%5D%2B%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%5C%22%2C%5C%22versionDetection%5C%22%3Atrue%2C%5C%22releaseDateAsVersion%5C%22%3Afalse%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22tutanota-app%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Atrue%2C%5C%22appName%5C%22%3A%5C%22%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Afalse%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22An%20email%20service%20with%20a%20strong%20focus%20on%20security%20and%20privacy%20that%20lets%20you%20encrypt%20emails%2C%20contacts%20and%20calendar%20entries%20on%20all%20your%20devices.%5C%22%2C%5C%22refreshBeforeDownload%5C%22%3Afalse%7D%22%2C%22overrideSource%22%3A%22GitHub%22%7D)
- App Source URL: `https://github.com/tutao/tutanota`
- Override Source: `None` or `GitHub`
- Trim Version String With ReGex: `[0-9]+\.[0-9]+\.[0-9]+`
- Filter APKs by Regular Expression: `tutanota-app`
- About: `An email service with a strong focus on security and privacy that lets you encrypt emails, contacts and calendar entries on all your devices.`

Tuta Calendar config:
Import App -> [Tuta Calendar](https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22de.tutao.calendar%22%2C%22url%22%3A%22https%3A%2F%2Fgithub.com%2Ftutao%2Ftutanota%22%2C%22author%22%3A%22tutao%22%2C%22name%22%3A%22Tuta%20Calendar%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22includePrereleases%5C%22%3Afalse%2C%5C%22fallbackToOlderReleases%5C%22%3Atrue%2C%5C%22filterReleaseTitlesByRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22filterReleaseNotesByRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22verifyLatestTag%5C%22%3Afalse%2C%5C%22dontSortReleasesList%5C%22%3Afalse%2C%5C%22useLatestAssetDateAsReleaseDate%5C%22%3Afalse%2C%5C%22releaseTitleAsVersion%5C%22%3Afalse%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22%5B0-9%5D%2B%5C%5C%5C%5C.%5B0-9%5D%2B%5C%5C%5C%5C.%5B0-9%5D%2B%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%5C%22%2C%5C%22versionDetection%5C%22%3Atrue%2C%5C%22releaseDateAsVersion%5C%22%3Afalse%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22calendar%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Atrue%2C%5C%22appName%5C%22%3A%5C%22%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Afalse%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22Quantum-safe%20encrypted%20planner%20to%20schedule%20%26%20manage%20your%20events%20%26%20sync%20calendars%20privately.%5C%22%2C%5C%22refreshBeforeDownload%5C%22%3Afalse%7D%22%2C%22overrideSource%22%3A%22GitHub%22%7D)
- App Source URL: `https://github.com/tutao/tutanota`
- Override Source: `None` or `GitHub`
- Trim Version String With ReGex: `[0-9]+\.[0-9]+\.[0-9]+`
- Filter APKs by Regular Expression: `calendar`
- About: `Quantum-safe encrypted planner to schedule & manage your events & sync calendars privately.`

As you can see, the only difference is the APK filter and the about description.

**Reply by glidingthrough** [reply]
Tuta Mail profile on Obtainium is as per the above instructions.

Installed version is more recent than "latest" detected by Obtainium - see attached file. Any ideas?
![Tuta](https://github.com/user-attachments/assets/63541f8e-9c75-421e-882c-8fa42e5a0a69)

**Reply by DarkCrypt** [reply]
There's no apk to download for 253.241203.1 within Tuta's GitHub releases so 253.241126.2 is still the latest. Obtainium will detect the next release when there's an apk. For now you can hit the checkmark and "Mark as Updated" to get rid of the update notification.

If you add the above Trim Version to your config it'll eliminate `tutanota-android-release-` from the version #.

Were you updating Tuta from the Play Store or elsewhere before using Obtainium? 253.241203.1 is on the Play Store.

**Comment by norbertkeri** [comment]
The calendar seems to have disappeared from the app list, or is it only me?

**Reply by DarkCrypt** [reply]
Searching through the PR's... looks like Tuta Calendar was never added, only Tuta Mail.
